lib/datadog/core/remote/client.rb

```diff
@@ -119,7 +119,7 @@ def payload
           client_tracer = {
             runtime_id: Core::Environment::Identity.id,
             language: Core::Environment::Identity.lang,
-            tracer_version: Core::Environment::Identity.tracer_version,
+            tracer_version: Core::Environment::Identity.tracer_version_semver2,
```

Comment:
I believe we should hoist the result value from it since running the regex for every request for remote configuration is going to be expensive

Reply:
Given the CI hell we're currently in I'll merge it as is and fix in a separate PR.

Reply:
PR: https://github.com/DataDog/dd-trace-rb/pull/2819